src/Node/Expression/Variable/AssignContextVariable.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Node\Expression\Variable;
+
+use Twig\Node\Expression\AssignNameExpression;
+
+final class AssignContextVariable extends AssignNameExpression
```

Comment:
should AssignContextVariable actually be an expression ? AFAIK, we don't support assigning context variables in arbitrary expressions (while the AST allows representing such case)

Reply:
We might support `{% if foo = some_expression %}...{% endif %}` at some point (I'm working on a new expression parser that would, at least, make it easy to support it).